tsl/src/remote/txn.c

```diff
@@ -269,23 +270,11 @@ exec_cleanup_command(TSConnection *conn, const char *query)
 			break;
 	}
 
-	if (success)
-	{
-		async_response_close(response);
-
-		/* that should have been the last response from the set */
-		response = async_request_set_wait_any_response_deadline(set, end_time);
-		Assert(response == NULL);
-	}
-	else
-	{
+	if (!success)
 		async_response_report_error(response, WARNING);
-		async_response_close(response);
 
-		/* drain the set until empty of all possibly queued errors */
-		while ((response = async_request_set_wait_any_response_deadline(set, end_time)))
-			async_response_close(response);
-	}
+	async_response_close(response);
+
```

Comment:
This hunk removes code that reaps the request queue, which is supposed to be handled by the `async_request_cleanup_result`, but if there is a bug this will not have happened. Is it possible to add some sort of check that there are no outstanding requests?

Reply:
As you mention, `async_request_cleanup_result` now handles the cleanup. Any assertions/checks that all results have been drained should happen in that function.

Reply:
The contract provided by libpq is that you should call `PGgetResult` until it returns NULL, which is what the drain function does. From the docs:

`
 A null pointer is returned when the command is complete and there will be no more results. PGgetResult must be called repeatedly until it returns a null pointer, indicating that the command is done.
`

I don't understand what the purpose is to double-check this, since we already reached the desired condition. It is a bit like asserting that a pointer is NULL in the if clause that checked for NULL. Further, it is not clear to me that any internal state of the connection would accurately reflect a "drained" connection, since the way to know it is drained is to call PGgetResult until it returns NULL, which we already do.